src/main/webapp/2Dscript.js

```diff
@@ -414,40 +476,67 @@ function placeRectangleEnd() {
   if (infoWindow !== undefined) {
     infoWindow.setMap(null);
   }
-  // Clear other paths.
-  // line
-  // minLatPoint
-  // lat , lng
-  // minLngPoint
-  // lat, lng
-  const subSectionData = computeSubSection(currentRun.data,
-      currentLat, priorLat, currentLng, priorLng);
-  // Choose a subSectionNumber, implement differently in future pr.
-  const subSectionNumber = 1;
-  // Clear prior paths, only display newly selected ones.
-  if (markerBottom !== undefined) {
-    markerBottom.setMap(null);
-  }
-  if (markerTop !== undefined) {
-    markerTop.setMap(null);
+  clearSelectedPaths(['subSection', 'markerBottom', 'markerTop']);
+  // Clear other paths. line minLatPoint lat , lng minLngPoint lat, lng Iterate
+  // over all data values here, need to find a way to check the checkbox value,
+  // should likely store this in an object somewhere to prevent excess DOM
+  // queries.
+  const mapRuns = Object.entries(runs);
+  for ([id, currentRun] of mapRuns) {
+    if (currentRun.checkBox.checked) {
```

Comment:
You can invert the conditional to reduce nesting:
```
if (!currentRun.checkBox.checked) {
  continue;
}
/* ... nominal-case code ... */
```
https://docs.google.com/document/d/16SSi-rIUGxJ5-ksifO6y0InCOwgbRa3lHBLnBJ8Klrk/edit?usp=sharing